### PR TITLE
Update dataset API with title/description fields

### PR DESCRIPTION
**Pull request by florimondmanca**
Refs #19 

Implem côté back d'une API exposant `POST /datasets/`, `GET /datasets/` et `GET /datasets/<id>` avec des champs `title, description` obligatoires.

cc @magopian Pour usage par le front :-)

### CODE REVIEW

File: tests/api/test_datasets.py [hunk 1]
```diff
@@ -3,24 +3,62 @@
...
+    data = response.json()
+    assert len(data["detail"]) == len(expected_errors_attrs)
+
+    for error, expected_error_attrs in zip(data["detail"], expected_errors_attrs):
+        error_attrs = {key: error[key] for key in expected_error_attrs}
+        assert error_attrs == expected_error_attrs
+
+
+@pytest.mark.asyncio
+async def test_dataset_create(client: httpx.AsyncClient) -> None:
```
Comment (magopian) [inline tests/api/test_datasets.py:34]: very minor nit: here the test is named `test_dataset_create`, while the previous one is `test_create_dataset...`. Maybe use the same naming convention?

**Review by magopian — APPROVED** [review]
One very minor nit, otherwise good to merge!
🚢 

Thanks @florimondmanca !

File: tests/api/test_datasets.py (hunk 1 above)
Comment (florimondmanca) [reply tests/api/test_datasets.py:34]: Oui bien vu ! C'est une coquille

```suggestion
async def test_create_dataset(client: httpx.AsyncClient) -> None:
```

**Comment by magopian** [comment]
Looks perfect, let's merge!

**Pull request merged**